test/lit/passes/simplify-locals-strings.wast

```diff
@@ -10,185 +10,9 @@
-  ;; CHECK-NEXT:    (i32.const 1)
-  ;; CHECK-NEXT:    (i32.const 2)
-  ;; CHECK-NEXT:   )
-  ;; CHECK-NEXT:  )
-  ;; CHECK-NEXT:  (i32.store
-  ;; CHECK-NEXT:   (i32.const 3)
-  ;; CHECK-NEXT:   (i32.const 4)
-  ;; CHECK-NEXT:  )
-  ;; CHECK-NEXT:  (drop
-  ;; CHECK-NEXT:   (local.get $temp)
-  ;; CHECK-NEXT:  )
-  ;; CHECK-NEXT:  (local.set $temp
-  ;; CHECK-NEXT:   (string.new_wtf8
-  ;; CHECK-NEXT:    (i32.const 1)
-  ;; CHECK-NEXT:    (i32.const 2)
-  ;; CHECK-NEXT:   )
-  ;; CHECK-NEXT:  )
-  ;; CHECK-NEXT:  (i32.store
-  ;; CHECK-NEXT:   (i32.const 3)
-  ;; CHECK-NEXT:   (i32.const 4)
-  ;; CHECK-NEXT:  )
-  ;; CHECK-NEXT:  (drop
-  ;; CHECK-NEXT:   (local.get $temp)
-  ;; CHECK-NEXT:  )
-  ;; CHECK-NEXT:  (local.set $temp
-  ;; CHECK-NEXT:   (string.new_lossy_utf8
-  ;; CHECK-NEXT:    (i32.const 1)
-  ;; CHECK-NEXT:    (i32.const 2)
-  ;; CHECK-NEXT:   )
-  ;; CHECK-NEXT:  )
-  ;; CHECK-NEXT:  (i32.store
-  ;; CHECK-NEXT:   (i32.const 3)
-  ;; CHECK-NEXT:   (i32.const 4)
-  ;; CHECK-NEXT:  )
-  ;; CHECK-NEXT:  (drop
-  ;; CHECK-NEXT:   (local.get $temp)
-  ;; CHECK-NEXT:  )
-  ;; CHECK-NEXT:  (local.set $temp
-  ;; CHECK-NEXT:   (string.new_wtf16
-  ;; CHECK-NEXT:    (i32.const 1)
-  ;; CHECK-NEXT:    (i32.const 2)
-  ;; CHECK-NEXT:   )
-  ;; CHECK-NEXT:  )
-  ;; CHECK-NEXT:  (i32.store
-  ;; CHECK-NEXT:   (i32.const 3)
-  ;; CHECK-NEXT:   (i32.const 4)
-  ;; CHECK-NEXT:  )
-  ;; CHECK-NEXT:  (drop
-  ;; CHECK-NEXT:   (local.get $temp)
-  ;; CHECK-NEXT:  )
-  ;; CHECK-NEXT: )
-  (func $no-new-past-store
-    (local $temp stringref)
-    ;; A string.new cannot be moved past a memory store.
-    (local.set $temp
-      (string.new_utf8
-        (i32.const 1)
-        (i32.const 2)
-      )
-    )
-    (i32.store
-      (i32.const 3)
-      (i32.const 4)
-    )
-    (drop
-      (local.get $temp)
-    )
-    (local.set $temp
-      (string.new_wtf8
-        (i32.const 1)
-        (i32.const 2)
-      )
-    )
-    (i32.store
-      (i32.const 3)
-      (i32.const 4)
-    )
-    (drop
-      (local.get $temp)
-    )
-    (local.set $temp
-      (string.new_lossy_utf8
-        (i32.const 1)
-        (i32.const 2)
-      )
-    )
-    (i32.store
-      (i32.const 3)
-      (i32.const 4)
-    )
-    (drop
-      (local.get $temp)
-    )
-    (local.set $temp
-      (string.new_wtf16
-        (i32.const 1)
-        (i32.const 2)
-      )
-    )
-    (i32.store
-      (i32.const 3)
-      (i32.const 4)
-    )
-    (drop
-      (local.get $temp)
-    )
-  )
-
-  ;; CHECK:      (func $yes-new-past-store (type $1)
-  ;; CHECK-NEXT:  (local $temp stringref)
-  ;; CHECK-NEXT:  (nop)
-  ;; CHECK-NEXT:  (drop
-  ;; CHECK-NEXT:   (i32.load
-  ;; CHECK-NEXT:    (i32.const 3)
-  ;; CHECK-NEXT:   )
-  ;; CHECK-NEXT:  )
-  ;; CHECK-NEXT:  (drop
-  ;; CHECK-NEXT:   (string.new_utf8
-  ;; CHECK-NEXT:    (i32.const 1)
-  ;; CHECK-NEXT:    (i32.const 2)
-  ;; CHECK-NEXT:   )
-  ;; CHECK-NEXT:  )
-  ;; CHECK-NEXT: )
-  (func $yes-new-past-store
```

Comment:
We don't seem to have any `yes-new-*` functions after this removal, so we might be losing coverage here.